src/sysroot.rs

```diff
@@ -153,12 +159,13 @@ version = "0.0.0"
         }
 
         // Copy artifacts to Xargo sysroot
+        let base_dir = match target_dir {
+            Some(dir) => config.unwrap().root().join(dir),
```

Comment:
If you fold the two matches into one here, the code becomes shorter and you avoid an `unwrap`.

Reply:
No, because I want to do `td.join("target")` if `config` is None **or** `config.target_dir()?` is None. So the only way I can think of to avoid the unwrap is to have nested matches and have `td.join("target")` twice. Not sure if that's better.

Reply:
Hm I see.
Either way please move `let target_dir` down to the other dir stuff here.

But then, wouldn't this work?
```rust
let base_dir = match config {
  Some(c) => c.target_dir()?.map(|d| c.root().join(d)),
  None => td.join("target"),
};
```

That's equivalent to your code, right?
However, it makes me wonder if this is really what we want. This directory will be outside the tempdir, right? We'll be spilling files into the user's dir and they might get mixed up with other stuff? Shouldn't this be more like `td.join(target_dir)` or so?

Reply:
Ah, good idea. Yes, that's equivalent and better, I'll use that.

From my point of view it's the desired behavior. Without setting the target-dir, the files are put into a temp directory. With the target-dir, the files are put into that directory, relative to the root directory of the project. So for example, you can set it to `"build/sysroot"` and then all files go into `"<yourproject>/build/sysroot"`.

Reply:
What is target-dir usually used for? Isn't that to change `target` to something else? So as a user I would expect by projects stuff to be put there, but not the sysroot.

Reply:
hm, I created the patch quite some time ago, but I remember that without target-dir, Xargo was putting all kinds of things in the usual target directory. I just tested it again and actually, using a temp directory for the sysroot and using `CARGO_TARGET_DIR` to store the project stuff at a different place, looks fine. 

Maybe Xargo behaves differently now or I'm missing something. However, I would suggest that we close the PR for now and I report back if there still is a good reason for target-dir.

Reply:
All right. Thanks a lot for looking into this!